### PR TITLE
feat: smart escrow checker

### PR DESCRIPTION
**Pull request by tynes**
**Description**

A simple script to ensure that the `SmartEscrow` contract has been
configured correctly.

Run the command `just check` to check

### CODE REVIEW

File: script/smart-escrow/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# SmartEscrow
+
+Checks to ensure that the Optimism + Base scart escrow contract is configured correctly.
```
Comment (mds1) [inline script/smart-escrow/README.md:3]:
```suggestion
Checks to ensure that the Optimism + Base [smart escrow contract](https://optimistic.etherscan.io/address/0x143F5773CFE5613ca94196d557c889134F47CB77) is configured correctly.
```

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+/// @notice https://optimistic.etherscan.io/address/0x143F5773CFE5613ca94196d557c889134F47CB77
+///         Ensure SmartEscrow is configured correctly
+contract SmartEscrowCheck is Script {
+  SmartEscrow constant SMART_ESCROW = SmartEscrow(0x143F5773CFE5613ca94196d557c889134F47CB77);
+
+  function run() public view {
+    require(block.chainid == 10, "Only OP Mainnet");
+
+    // OP Token is configured correctly
+    require(SMART_ESCROW.OP_TOKEN() == 0x4200000000000000000000000000000000000042);
```
Comment (mds1) [inline script/smart-escrow/SmartEscrowCheck.s.sol:41]: Can we add reason strings to all the require statements? If a check fails, it's much easier to determine which one failed when there is a unique revert reason. Even just incrementing numbered strings like `"100"`, `"200"`, etc. is sufficient to find the failing line (similar to the gnosis safe [approach](https://github.com/safe-global/safe-smart-account/blob/12136fd54940892bf43258f86b6e8f9d0eff69ea/contracts/Safe.sol#L313))

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    // 2 of 2 multisig
+    require(SMART_ESCROW.owner() == 0x0a7361e734cf3f0394B0FC4a45C74E7a4Ec70940);
+    require(SMART_ESCROW.defaultAdmin() == 0x0a7361e734cf3f0394B0FC4a45C74E7a4Ec70940);
```
Comment (mds1) [inline script/smart-escrow/SmartEscrowCheck.s.sol:45]: Lots of hardcoded addresses, would be great to pull these out into constants so the var name can describe each

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    GnosisSafeL2 owner = GnosisSafeL2(SMART_ESCROW.owner());
+
+    address[] memory owners = owner.getOwners();
+    // First owner is optimism multisig
+    require(owners[0] == 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0);
+    // Second owner is base multisig
+    require(owners[1] == 0x6e1DFd5C1E22A4677663A81D24C6BA03561ef0f6);
```
Comment (mds1) [inline script/smart-escrow/SmartEscrowCheck.s.sol:52]: Probably should also check the length of owners + Safe threshold

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    require(SMART_ESCROW.benefactor() == 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0);
+
+    // Beneficiary is an eoa
+    require(SMART_ESCROW.beneficiary() == 0x635Fb974F09B269Bc750bF96338c29cF41430125);
+
+    require(SMART_ESCROW.start() == 1720674000);
+    require(SMART_ESCROW.end() == 1878462000);
+    require(SMART_ESCROW.vestingPeriod() == 7889400);
+    require(SMART_ESCROW.initialTokens() == 17895697000000000000000000);
+    require(SMART_ESCROW.vestingEventTokens() == 4473924000000000000000000);
```
Comment (mds1) [inline script/smart-escrow/SmartEscrowCheck.s.sol:74]: Would be nice if there was some docs etc we can link to as a source for these numbers, expected beneficiary, etc

**Review by mds1 — COMMENTED** [review]
LGTM overall, mainly just comments around making things a bit more clear and self-documenting

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    // Benefactor is optimism multisig
+    require(SMART_ESCROW.benefactor() == 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0);
+
+    // Beneficiary is an eoa
+    require(SMART_ESCROW.beneficiary() == 0x635Fb974F09B269Bc750bF96338c29cF41430125);
+
+    require(SMART_ESCROW.start() == 1720674000);
+    require(SMART_ESCROW.end() == 1878462000);
+    require(SMART_ESCROW.vestingPeriod() == 7889400);
+    require(SMART_ESCROW.initialTokens() == 17895697000000000000000000);
```
Comment (tynes) [inline script/smart-escrow/SmartEscrowCheck.s.sol:73]: Initial tokens and vesting event tokens are correct

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    // Benefactor is optimism multisig
+    require(SMART_ESCROW.benefactor() == 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0);
+
+    // Beneficiary is an eoa
+    require(SMART_ESCROW.beneficiary() == 0x635Fb974F09B269Bc750bF96338c29cF41430125);
+
+    require(SMART_ESCROW.start() == 1720674000);
+    require(SMART_ESCROW.end() == 1878462000);
+    require(SMART_ESCROW.vestingPeriod() == 7889400);
```
Comment (tynes) [inline script/smart-escrow/SmartEscrowCheck.s.sol:72]: vesting period is about 3 months

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    require(SMART_ESCROW.hasRole(SMART_ESCROW.TERMINATOR_ROLE(), 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0));
+    require(SMART_ESCROW.hasRole(SMART_ESCROW.TERMINATOR_ROLE(), 0x6e1DFd5C1E22A4677663A81D24C6BA03561ef0f6));
+
+    // Benefactor is optimism multisig
+    require(SMART_ESCROW.benefactor() == 0x2501c477D0A35545a387Aa4A3EEe4292A9a8B3F0);
+
+    // Beneficiary is an eoa
+    require(SMART_ESCROW.beneficiary() == 0x635Fb974F09B269Bc750bF96338c29cF41430125);
+
+    require(SMART_ESCROW.start() == 1720674000);
```
Comment (tynes) [inline script/smart-escrow/SmartEscrowCheck.s.sol:70]: start is `Wed Jul 10 2024 23:00:00 GMT-0600 (Central Standard Time)`

File: script/smart-escrow/SmartEscrowCheck.s.sol [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    require(SMART_ESCROW.end() == 1878462000);
```
Comment (tynes) [inline script/smart-escrow/SmartEscrowCheck.s.sol:71]: end is `Wed Jul 11 2029 05:00:00 GMT-0600 (Central Standard Time)`

Comment (tynes) [inline script/smart-escrow/SmartEscrowCheck.s.sol:71]: end is `Wed Jul 11 2029 05:00:00 GMT-0600 (Central Standard Time)`

**Comment by tynes** [comment]
I am fine with closing this PR

**Pull request closed without merging**